lib/Elastica/Bulk/ResponseSet.php

```diff
@@ -95,13 +95,11 @@ public function hasError()
     }
 
     /**
-     * @return bool|\Elastica\Bulk\Response
+     * @return \Elastica\Bulk\Response
      */
     public function current()
     {
-        return $this->valid()
-            ? $this->_bulkResponses[$this->key()]
-            : false;
+        return $this->_bulkResponses[$this->key()];
```

Comment:
I wonder if we should treat this as a bug or breaking change ... Someone re

Reply:
I hope nobody would be affected by this potential bc break as it means people would use the Iterator in a wrong way. As an alternative we could leave the implementation and only change the phpdoc. But that would also be confusing.

Reply:
It reminds me of https://github.com/ruflin/Elastica/issues/1374 Let's ping people in there on what they think.

Reply:
I'd take the pragmatic approach here - it could only break code that was already abusing a well-defined mechanism, and I can't think of a really practical example how it would *hurt* existing code, it would more probably *fix* it by accident. So I wouldn't call BC on this.

Reply:
Looks indeed like a case of xkcd workflow ![xkcd workflow](https://imgs.xkcd.com/comics/workflow.png)

Reply:
What would we do without xkcd?

Sounds like we are all agreement we should move forward with this.

@Tobion I suggest we still add an entry to the breaking changes for people to be aware of it.

Reply:
@ruflin done. This is ready from my point of view.